### DISCUSSION TITLE
posts/2022/04/static-tweets-astro/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#4] [Announcements]
# posts/2022/04/static-tweets-astro/

A component which properly embeds tweets when you’re using today’s hottest SSG.

https://www.brycewray.com/posts/2022/04/static-tweets-astro/

**Comment by davidsneighbour** [comment]
Do you have an opinion about the "staticness" of these approaches and the deletion of tweets or the new rumoured "edit" feature? I checked the Twitter Dev docs and they are relatively foggy about what to do in case of updates or deletions. Assuming I might update my site once in a while it might be probably best to have some form of indicator of the last update near the tweet or some GDPR-like legal small print? Check "III. Updates and Removals" on https://developer.twitter.com/en/developer-terms/agreement

**Reply by brycewray** [reply]
I’ve been tied up with family activities over the last couple of days, so I haven’t had much chance to read up on this new feature, although I did catch that it was reported. Perhaps the `created_at` parameter will be changed to account for this, **or** Twitter will add a new `last_edited_at` parameter. I guess we will have to wait for something official from Big T. But, yes, I would think it would be good to add that, too, however it ends up being manifested.

**Comment by mettamyron** [comment]
Thanks for sharing this. Do you know how I could get the like_count for the tweet? I tried this, but it's not working.

```
like_count = JsonData.public_metrics[0].like_count
```

**Reply by brycewray** [reply]
@maitrimyron Have you tried `JsonData.public_metrics.like_count` (without the `[0]`)?

(Incidentally, the [official astro-embed plugin](https://github.com/astro-community/astro-embed) uses the older Public Syndication API, where it’s much simpler: a top-level `favorite_count`.)

**Reply by mettamyron** [reply]
I have Bryce. I think my request URL is missing the inclusion, but I can't work it out.
```
const jsonURL2 = "&expansions=author_id,attachments.media_keys&tweet.fields=created_at,text,attachments,entities,source,public_metrics&user.fields=name,username,profile_image_url&media.fields=preview_image_url,type,url,alt_text"
```
`like_count` is nested under `public_metrics` then, but how do I include it in the call?

**Reply by brycewray** [reply]
@maitrimyron I think you’re right. Sorry I didn‘t think of that the first time. Try changing that line to the following, which will give you pretty much the whole enchilada for the V2 API (or, to be sure, `public_metrics`):

```js
const jsonURL2 = "&expansions=author_id,referenced_tweets.id,in_reply_to_user_id,geo.place_id,attachments.media_keys,attachments.poll_ids,entities.mentions.username,referenced_tweets.id.author_id&tweet.fields=id,created_at,text,author_id,in_reply_to_user_id,referenced_tweets,attachments,withheld,geo,entities,public_metrics,possibly_sensitive,source,lang,context_annotations,conversation_id,reply_settings&user.fields=id,created_at,name,username,protected,verified,withheld,profile_image_url,location,url,description,entities,pinned_tweet_id,public_metrics&media.fields=media_key,duration_ms,height,preview_image_url,type,url,width,public_metrics,alt_text&place.fields=id,name,country_code,place_type,full_name,country,contained_within,geo&poll.fields=id,options,voting_status,end_datetime,duration_minutes"
```

**Reply by mettamyron** [reply]
Thanks @brycewray. It still stubbornly refuses to comply with my demands. For now I have kludged together something from v1.1 using`twitter-api-client` and your v2 code. Hoping I can figure it out before the old API is deprecated.

**Reply by brycewray** [reply]
> Thanks @brycewray. It still stubbornly refuses to comply with my demands. For now I have kludged together something from v1.1 using`twitter-api-client` and your v2 code. Hoping I can figure it out before the old API is deprecated.

@maitrimyron Sorry it didn‘t work, although now I really don‘t understand why. In the meantime, that Astro plugin I mentioned definitely will do the job. (Given the current limitations of the V2 API and its much more complicated structure, I'm no longer using it myself but, instead, have reverted to that same hoary old Public Syndication API on which the Astro plugin relies. It's so much easier to use and provides all the data that one is likely to need.)
